### ISSUE TITLE
Missing docstrings in V4

### ISSUE THREAD

**Issue by StijnKas** [pegasystems/pega-datascientist-tools#280]
### pdstools version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/pdstools/) of pdstools.

### Issue description

Creating issue for tracking purposes. While refactoring for the V4 version of pdstools, not every class/module/method/function was given docstrings. Now that I've included numpydoc as a pre-commit hook, this becomes an issue. I'd like to use numpydoc to validate new incoming docstrings automatically, but we can't do that today as it's complaining about many missing docstrings instead. This'll be quite some work, but obviously it's necessary to get the API docs up to speed as well.

### Reproducible example

```Python
-
```

### Expected behavior

Unfortunately (for us), **every** class, method, function should have a docstring, conforming exactly to the numpy docstring standard. I say unfortunately, because this is a lot of work.

### Installed versions

<details>

```
V4
```

</details>

**Comment by operdeck** [comment]
Should we really? It’s an internal tool with not that many users. I rather
spend time to broaden the scope of the analyses and make them better - eg
AGB or IH coverage or the ADM Healthcheck now that is going to be served
via DJS.

Op vr 8 nov 2024 om 10:40 schreef Stijn Kas ***@***.***>

**Comment by StijnKas** [comment]
@operdeck  I don't believe they are exclusive and I think there may be some middle-ground (only requiring valid docstrings for _new_ code), but I do think there's quite some benefits to having proper documentation so that it's more self-documented. But definitely good to not do docstrings for the sake of docstrings, so maybe this issue is worded a bit strongly.
